components/provisioner/internal/operations/executor.go

```diff
@@ -115,6 +115,10 @@ func (e *Executor) process(operation model.Operation, cluster model.Cluster, log
 
 		result, err := step.Run(cluster, operation, log)
 		if err != nil {
+			if errors.Is(err, errors.New("cluster kubeconfig is nil")) {
```

Comment:
If this is a known error with a const message, wouldn't it make sense to create it once, export it and reuse from variable?

Example from Go's std:
```go
var (
	ErrInvalid    = errors.New("invalid argument")
	ErrPermission = errors.New("permission denied")
	ErrExist      = errors.New("file already exists")
	ErrNotExist   = errors.New("file does not exist")
	ErrClosed     = errors.New("file already closed")
)

Reply:
Benefits of this solution would be visible with any changes around this error in the future. From changing anything about the error (message, structure), to finding all places where given error is mentioned with refactoring tools

Reply:
So to summarise, it should look something like this:
```go
var ErrKubeconfigNil = errors.New("cluster kubeconfig is nil")

func (s *CreateBindingsForOperatorsStep) Run(cluster model.Cluster, _ model.Operation, log logrus.FieldLogger) (operations.StageResult, error) {
	if cluster.Kubeconfig == nil {
		return operations.StageResult{}, ErrKubeconfigNil
	}
	// ...
}

func (e *Executor) process(operation model.Operation, cluster model.Cluster, logger logrus.FieldLogger) (bool, time.Duration, error) {
  //...
  if err != nil {
	  if errors.Is(err, ErrKubeconfigNil) {
		  log.Warnf("Warning, the %s", err.Error())
		  break
	  }
          log.Warnf("error while processing operation, stage failed: %s", err)
          return false, 0, err
  }
  //...
}
```

Reply:
This is the one time error in code. In my opinion in this fix, it is not necessary to export a single variable. It might be a great idea to refactor the way of logging in the whole component.

Reply:
IMO it might make sense to refactor the area being worked on instead of one big refactor task

Reply:
I agree with @VOID404 . We can apply this change as this is a good practice.

Reply:
May I ask why break when errors.Is(err, ErrKubeconfigNil) in executor? After break, it will go to e.updateOperationStatus(logger, operation.ID, "Operation succeeded", model.Succeeded, time.Now()), then if ErrKubeconfigNil, the provision operation will be considered as succeeded?